pkg/service/repair/helper_integration_test.go

```diff
@@ -0,0 +1,467 @@
+	}
+	if sched.id == "" {
+		t.Error("Not fully initialized repair sched resp")
+		return repairSchedResp{}
+	}
+
+	return sched
+}
+
+func newRepairStatusResp(t *testing.T, resp *http.Response) (repairStatusResp, bool) {
+	if resp.StatusCode != http.StatusOK {
+		return repairStatusResp{}, false
+	}
+	if isOldRepairStatusReq(resp.Request) {
+		return newOldRepairStatusResp(t, resp), true
+	}
+	return repairStatusResp{}, false
+}
+
+func newOldRepairStatusResp(t *testing.T, resp *http.Response) repairStatusResp {
+	req, ok := newRepairStatusReq(t, resp.Request)
+	if !ok {
+		t.Error("Not repair status resp")
+		return repairStatusResp{}
+	}
+
+	body := string(copyRespBody(t, resp))
+	var status repairStatus
+	switch scyllaclient.CommandStatus(strings.Trim(body, "\"")) {
+	case scyllaclient.CommandSuccessful:
+		status = repairStatusDone
+	case scyllaclient.CommandFailed:
+		status = repairStatusFailed
+	case scyllaclient.CommandRunning:
+		status = repairStatusRunning
+	default:
+		t.Error("Unknown old repair status: " + body)
+		return repairStatusResp{}
+	}
+
+	return repairStatusResp{
+		repairStatusReq: req,
+		status:          status,
+	}
+}
+
+func mockRepairSchedRespBody(t *testing.T, req *http.Request) (io.ReadCloser, bool) {
+	if isOldRepairSchedReq(req) {
+		return mockOldRepairSchedRespBody(t, req), true
+	}
+	return nil, false
+}
+
+var repairTaskCounter int32
+
+func mockOldRepairSchedRespBody(t *testing.T, req *http.Request) io.ReadCloser {
+	if !isOldRepairSchedReq(req) {
+		t.Error("Not old repair sched req")
+	}
+
+	return io.NopCloser(bytes.NewBufferString(fmt.Sprint(atomic.AddInt32(&repairTaskCounter, 1))))
+}
+
+func mockRepairStatusRespBody(t *testing.T, req *http.Request, status repairStatus) (io.ReadCloser, bool) {
+	if isOldRepairStatusReq(req) {
+		return mockOldRepairStatusRespBody(t, req, status), true
+	}
+	return nil, false
+}
+
+func mockOldRepairStatusRespBody(t *testing.T, req *http.Request, status repairStatus) io.ReadCloser {
+	if !isOldRepairStatusReq(req) {
+		t.Error("Not old repair status req")
+		return nil
+	}
+
+	var s scyllaclient.CommandStatus
+	switch status {
+	case repairStatusDone:
+		s = scyllaclient.CommandSuccessful
+	case repairStatusFailed:
+		s = scyllaclient.CommandFailed
+	case repairStatusRunning:
+		s = scyllaclient.CommandRunning
+	default:
+		t.Errorf("Unknown old repair status: %d", status)
+		return nil
+	}
+
+	return io.NopCloser(bytes.NewBufferString(fmt.Sprintf("%q", s)))
+}
+
+func repairStatusInterceptor(t *testing.T, status repairStatus) http.RoundTripper {
+	return httpx.RoundTripperFunc(func(req *http.Request) (*http.Response, error) {
+		if body, ok := mockRepairSchedRespBody(t, req); ok {
+			resp := httpx.MakeResponse(req, http.StatusOK)
+			resp.Body = body
+			return resp, nil
+		}
+		if body, ok := mockRepairStatusRespBody(t, req, status); ok {
+			resp := httpx.MakeResponse(req, http.StatusOK)
+			resp.Body = body
+			return resp, nil
+		}
+		return nil, nil
+	})
+}
```

Comment:
I have a bit of the problem with this function.
The name states (at least that's how I interpret it):
- I'm creating the interceptor that filters out only RepairStatusEndpoint calls 

I don't know what it does after intercepting it.
I must go to the function body to understand that it's not only intercepting repair_status but repair_async as well.

Why is that ? What is the rationale for that implementation ?